text/0000-build-std/4-build-std-always.md

```diff
@@ -0,0 +1,1257 @@
+
+This is aimed at supporting the following [motivations](./3-motivation.md):
+
+- Building the standard library on a stable toolchain without Cargo
+- Building standard library crates that are not shipped for a target
+- Using the standard library with tier three targets
+
+# Proposal
+[proposal]: #proposal
+
+Cargo configuration will contain a new key `build-std` under the `[build]`
+section ([?][rationale-build-std-in-config]), permitting one of two values -
+"never" ([?][rationale-build-std-never]) or "always", defaulting to "never":
+
+```toml
+[build]
+build-std = "always" # or `never`
+```
+
+`build-std` can also be specified in the `[target.<triple>]` and
+`[target.<cfg>]` sections ([?][rationale-build-std-target-section]):
+
+```toml
+[target.aarch64-unknown-illumos]
+build-std = "always" # or `never`
+```
+
+The `build-std` configuration locations have the following precedence
+([?][rationale-build-std-precedence]):
+
+1. `[target.<triple>]`
+2. `[target.<cfg>]`
+3. `[build]`
+
+As the Cargo configuration is local to the current user (typically in
+`.config/cargo.toml` in the project root and/or Cargo home directory), the value
+of `build-std` is not influenced by the dependencies of the current crate.
+
+When `build-std` is set to "always", then the standard library will be
+unconditionally recompiled ([?][rationale-unconditional]) in the release profile
+defined in its workspace as part of every clean build
+([?][rationale-release-profile]). This is primarily useful for users of tier
+three targets. As with other dependencies, the standard library's build will
+respect the `RUSTFLAGS` environment variable.
+
+> [!NOTE]
+>
+> Configuration of the pre-built standard library is split across bootstrap and
+> the Cargo packages for the standard library. As much of this configuration as
+> possible should be moved to the Cargo profile for these packages so that the
+> artifacts produced by build-std match the pre-built standard library as much
+> as is feasible.
+
+Alongside `build-std`, a `build-std-crate` key will be introduced
+([?][rationale-build-std-crate]), which can be used to specify which crates from
+the standard library should be built. Only "core", "alloc" and "std" are valid
+values for `build-std-crate`.
+
+```toml
+[build]
+build-std-crate = "std"
+```
+
+A value of "std" means that every crate in the graph has a direct dependency on
+`std`, `alloc` and `core`. Similarly, "alloc" means `alloc` and `core`, and
+"core" means just `core`.
+
+If [*Standard library dependencies*][deps] are implemented then `builtin`
+dependencies will be used if `build-std-crates` is not explicitly set.
+Otherwise, `build-std-crate` will default to the crate intended to be supported
+by the target (see later
+[*Target standard library support*][target-standard-library-support] section).
+
+If `std` is to be built and Cargo is building a test using the default test
+harness then Cargo will also build the `test` crate.
```

Comment:
I'd recommend moving this 1 paragraph up, I missed this on the current read through and was left confused. Including this in the same place we discuss what the available options are seems to read more naturally to me.

Also, we probably mean 'building a test or a benchmark' here?

Reply:
Reworded and reordered in 3214dbae